src/lib/Microsoft.Health.Fhir.Ingest/Service/NormalizationDataMappingException.cs

```diff
@@ -0,0 +1,37 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Text;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class NormalizationDataMappingException : Exception
+    {
+        public NormalizationDataMappingException(Exception ex)
+            : base(BuildMessage(ex), ex)
+        {
+            EnsureArg.IsNotNull(ex, nameof(ex));
+        }
+
+        private static string BuildMessage(Exception ex)

```

Comment:
Sounds good.  I was hesitant to do as a public extension because I wasn't sure how useful it would be but if you would like to use in your PR/changes please feel free to refactor and put it in a public extensions.